src/libraries/Microsoft.Extensions.Logging/src/LoggerMessageAttribute.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Provides information to guide the production of a strongly-typed logging method.
+    /// </summary>
+    /// <remarks>
+    /// The method this attribute is applied to:
+    ///    - Must be a partial method.
+    ///    - Must return <c>void</c>.
+    ///    - Must not be generic.
+    ///    - Must have an <see cref="ILogger"/> as one of its parameters.
+    ///    - Must have a <see cref="Microsoft.Extensions.Logging.LogLevel"/> as one of its parameters.
+    ///    - None of the parameters can be generic.
+    /// </remarks>
+    /// <example>
+    /// static partial class Log
+    /// {
+    ///     [LoggerMessage(EventId = 0, Message = "Could not open socket for {hostName}")]
+    ///     static partial void CouldNotOpenSocket(ILogger logger, LogLevel level, string hostName);
+    /// }
+    /// </example>
```

Comment:
I've never seen an `<example>` section render nicely in MS Docs. For example, look at `LoggerExtensions.LogTrace` - the [MS Docs page](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.logtrace?view=dotnet-plat-ext-5.0#Microsoft_Extensions_Logging_LoggerExtensions_LogTrace_Microsoft_Extensions_Logging_ILogger_System_String_System_Object___) shows the code without syntax highlighting but in monospace font, but only because the [xml code itself](https://github.com/dotnet/dotnet-api-docs/blob/26e6fe413e3c09929b016c28158589d9122b2ca9/xml/Microsoft.Extensions.Logging/LoggerExtensions.xml#L1069-L1071) was wrapped in `<code class="lang-csharp">`.

A better way to make sure the code renders correctly is to move the example to the remarks section, and you need to add it as markdown, like this:

```cs
/// <remarks>
/// The method this attribute is applied to:
/// ...
/// ...
/// <format type="text/markdown"><![CDATA[
/// ## Example
/// ```csharp
/// static partial class Log
/// {
///     [LoggerMessage(EventId = 0, Message = "Could not open socket for {hostName}")]
///     static partial void CouldNotOpenSocket(ILogger logger, LogLevel level, string hostName);
/// }
/// ```
/// ]]></format>
/// </remarks>
```

Reply:
I'll take care of this in a next PR

Reply:
We noticed the behavior of intellisense is to show both class summary and class remarks in the intellisense dropdown when hovering on the attribute, so the above suggestion is incorrect.

A better way to add an example and ensure it shows up in MS Docs with proper highlighting, is to still add the `<example>` section but with the code wrapped in markdown:

```cs
    /// <example>
    /// <format type="text/markdown"><![CDATA[
    /// ```csharp
    /// static partial class Log
    /// {
    ///     [LoggerMessage(EventId = 0, Message = "Could not open socket for {hostName}")]
    ///     static partial void CouldNotOpenSocket(ILogger logger, LogLevel level, string hostName);
    /// }
    /// ```
    /// ]]></format>
    /// </example>
```